sklearn/linear_model/tests/test_logistic.py

```diff
@@ -2065,6 +2076,47 @@ def test_liblinear_not_stuck():
         clf.fit(X_prep, y)
 
 
+def test_lr_cv_scores_differ_when_sample_weight_is_requested():
+    rng = np.random.RandomState(10)
+    X, y = make_classification(n_samples=10, random_state=rng)
+    sample_weight = np.ones(len(y))
+    sample_weight[: len(y) // 2] = 2
+    kwargs = {"sample_weight": sample_weight}
+
+    with config_context(enable_metadata_routing=True):
+        scorer1 = get_scorer("accuracy")
+        lr_cv1 = LogisticRegressionCV(scoring=scorer1)
+        lr_cv1.fit(X, y, **kwargs)
+
+        scorer2 = get_scorer("accuracy")
+        scorer2.set_score_request(sample_weight=True)
+        lr_cv2 = LogisticRegressionCV(scoring=scorer2)
+        lr_cv2.fit(X, y, **kwargs)
+
+    assert pytest.approx(lr_cv1.scores_[1]) != lr_cv2.scores_[1]
+
```

Comment:
Same here, we should be testing `model.score`.

Reply:
These tests are basically testing for the routing in the fit method. I am not sure if the score method would return exactly the same results.

Reply:
What I mean here is the following: we create a scorer that requests sample weights. We should then be sure that the calling score will request sample weights and that the score is different from the one not requesting sample weights.

The reason is that the `score` method will use the `scorer` pass to the `__init__` if I am not wrong.

Reply:
Added a test for this.